### PR TITLE
Add build components to Chimera

### PR DESCRIPTION
**Pull request by Szymongib**
#### Summary
This PR adds build components to Chimera.
- Added Dockerfile
- Extended Makefile
- Added scripts for CI docker push
- Added CI config

### CODE REVIEW

**Review by stafot — APPROVED** [review]
Nice work! You could also add some functionality regarding cut releasing automation through circleCI.

File: scripts/push-docker-pr.sh [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+set -e
+set -u
+
+export TAG="${CIRCLE_SHA1:0:7}"
+
+echo $DOCKER_PASSWORD | docker login --username $DOCKER_USERNAME --password-stdin
+
+docker tag mattermost/chimera:test mattermost/chimera:$TAG
+
+docker push mattermost/chimera:$TAG
```
Comment (gabrieljackson) [inline scripts/push-docker-pr.sh:15]: nit: missing newline but 0/5

File: scripts/push-docker.sh [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+else
+  echo "Pushing release $CIRCLE_TAG..."
+  TAG="$CIRCLE_TAG"
+fi
+
+echo $DOCKER_PASSWORD | docker login --username $DOCKER_USERNAME --password-stdin
+
+docker tag mattermost/chimera:test mattermost/chimera:$TAG
+
+docker push mattermost/chimera:$TAG
```
Comment (gabrieljackson) [inline scripts/push-docker.sh:21]: same here

**Review by gabrieljackson — APPROVED** [review]
LGTM!

**Pull request merged**